src/realm/decimal128.cpp

```diff
@@ -319,6 +319,12 @@ bool Decimal128::is_null() const
     return m_value.w[0] == 0xaa && m_value.w[1] == 0x7c00000000000000;
 }
 
+bool Decimal128::is_nan() const
```

Comment:
Intel does provide their own implementation of this function, but it requires pulling in a chain of other files including the their `from_string` function which I think we wanted to avoid for some reason?